hieroplatform/src/main/java/org/hiero/sketch/table/BaseRowSnapshot.java

```diff
@@ -0,0 +1,89 @@
+package org.hiero.sketch.table;
+
+import org.hiero.sketch.table.api.IRow;
+import org.hiero.utils.Converters;
+
+public abstract class BaseRowSnapshot implements IRow {
+    protected final Schema schema;
+
+    protected BaseRowSnapshot(Schema schema) {
+        this.schema = schema;
+    }
+
+    @Override
+    public int rowSize() {
+        return this.schema.getColumnCount();
+    }
+
+    @Override
+    public Schema getSchema() {
+        return this.schema;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
How about adding here a compareTo method between two BaseRowSnapshots?
This should enable you to remove a bunch of code from many other places.